python/ray/serve/_private/proxy.py

```diff
@@ -288,19 +291,21 @@ def update_draining(self, draining: bool):
             self._draining_start_time = None
 
     @abstractmethod
-    async def not_found(self, proxy_request: ProxyRequest) -> ResponseGenerator:
-        raise NotImplementedError
-
-    @abstractmethod
-    async def draining_response(self, proxy_request: ProxyRequest) -> ResponseGenerator:
```

Comment:
`draining_response` was actually used for both routes and health responses, which was quite confusing. I've instead just passed a `healthy` bit into those existing methods.

Reply:
Love how this works now! Thanks for the improvement, Ed!